### PR TITLE
Add mlflow_logging param

### PR DESCRIPTION
**Pull request by garar**
## Why are these changes needed?

I've added a parameter, that allows for disabling MLflow logging. Currently FLAML does log all runs to MLflow if MLflow is installed and a run is started. I'm using my own MLflow logging, and I would like to disable built-in FLAML MLflow logging. There are also, some other users asking for this: https://github.com/microsoft/FLAML/issues/880#issuecomment-1478883732

This is my first FLAML PR, so please let know if anything is missing. I didn't add any tests, because there are no tests for MLFlow integration, except some notebooks. If you think I should add new tests, please let me know.

## Related issue number

There is no issue that is directly related to my PR, if I need to create one please let me know.

## Checks

- [x] I've used [pre-commit](https://microsoft.github.io/FLAML/docs/Contribute#pre-commit) to lint the changes in this PR (note the same in integrated in our CI checks).
- [x] I've included any doc changes needed for https://microsoft.github.io/FLAML/. See https://microsoft.github.io/FLAML/docs/Contribute#documentation to build and test documentation locally.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

**Comment by garar** [comment]
@microsoft-github-policy-service agree

File: flaml/automl/automl.py [hunk 1]
```diff
@@ -1659,6 +1669,7 @@ def cv_score_agg_func(val_loss_folds, log_metrics_folds):
         self._state.fit_kwargs = fit_kwargs
         custom_hp = custom_hp or self._settings.get("custom_hp")
         self._skip_transform = self._settings.get("skip_transform") if skip_transform is None else skip_transform
+        self._mlflow_logging = mlflow_logging or self._settings.get("mlflow_logging")
```
Comment (sonichi) [inline flaml/automl/automl.py:1672]:
```suggestion
        self._mlflow_logging = self._settings.get("mlflow_logging") if mlflow_logging is None else mlflow_logging
```

**Review by sonichi — APPROVED** [review]
Looks good to me. One optional suggestion is to add `mlflow_logging=False` in one of the existing tests, to make sure this feature works as intended in future.

File: flaml/automl/automl.py (hunk 1 above)
Comment (garar) [reply flaml/automl/automl.py]: Fixed.

**Comment by garar** [comment]
Hi @sonichi!

Thank you for your comment, I've applied your suggestions and I added tests for this code.

I see that Python 3.8 test for my PR are failing, I wonder if there is something wrong with them because they are also failing for other PRs: Example: https://github.com/microsoft/FLAML/actions/runs/4836511748

Should I worry about these tests or is it there some instability?

Thanks again!

**Comment by sonichi** [comment]
> Hi @sonichi!
> 
> Thank you for your comment, I've applied your suggestions and I added tests for this code.
> 
> I see that Python 3.8 test for my PR are failing, I wonder if there is something wrong with them because they are also failing for other PRs: Example: https://github.com/microsoft/FLAML/actions/runs/4836511748
> 
> Should I worry about these tests or is it there some instability?
> 
> Thanks again!

Thanks for adding the tests. 
@thinkall could you take a look at the failures? They are related to spark.

File: website/docs/Use-Cases/Task-Oriented-AutoML.md [hunk 2]
```diff
@@ -477,6 +477,18 @@ with mlflow.start_run():
     automl.fit(X_train=X_train, y_train=y_train, **settings)
 ```
 
+To disable mlflow logging, set `mlflow_logging=False`:
```
Comment (thinkall) [inline website/docs/Use-Cases/Task-Oriented-AutoML.md:480]:
```suggestion
To disable mlflow logging pre-configured in FLAML, set `mlflow_logging=False`:
```

Comment (thinkall) [reply website/docs/Use-Cases/Task-Oriented-AutoML.md:480]: Would it reduce ambiguity to mention "pre-configured in FLAML"?

**Comment by thinkall** [comment]
> > Hi @sonichi!
> > Thank you for your comment, I've applied your suggestions and I added tests for this code.
> > I see that Python 3.8 test for my PR are failing, I wonder if there is something wrong with them because they are also failing for other PRs: Example: https://github.com/microsoft/FLAML/actions/runs/4836511748
> > Should I worry about these tests or is it there some instability?
> > Thanks again!
> 
> Thanks for adding the tests. @thinkall could you take a look at the failures? They are related to spark.

This is caused by ipython-8.13.0. But I see the new runs will install ipython-8.12.1 correctly for python 3.8

**Comment by garar** [comment]
Hi @sonichi @thinkall !

I've applied @thinkall comments. I've also seen that @thinkall fixed catboost failures on mac on main branch, so I've rebased my code. Can you rerun the workflows? Fingers crossed, everything passes and this PR will be ready for another review and merge.

Thanks!

**Review by thinkall — APPROVED** [review]

**Pull request merged**